content/blog/ultimate-windows-development-environment-guide/index.md

```diff
@@ -0,0 +1,612 @@
+
+```
+[diff]
+  tool = default-difftool
+[difftool "default-difftool"]
+  cmd = code --wait --diff $LOCAL $REMOTE
+```
+
+And it should take care of the rest for you.
+
+### Line Endings {#git-line-endings}
+
+While most high-level language code is interoperable between different OSes, one of the primary differences between high-level codebases in Windows vs. macOS or Linux is the line-endings. As you might know, Windows uses `\r\n` line-ending where Linux and macOS end with `\n`.
+Luckily for us, Git can automatically convert the Windows line-endings before committing them to the repository. To do so, simply run the following command:
+
+```
+git config --global core.autocrlf true
+```
+
+## WSL {#wsl}
+
+Alright, alright, I'm sure you've been expecting to see this here. I can't beat around the bush any longer. Windows Subset for Linux (WSL) enables users to run commands on a Linux instance without having to dual-boot or run a virtual-machine themselves. The way that it works differs from version to version. The initial version worked by mapping system calls from Windows to Linux in a somewhat complex method. The new version (WSL2), right around the corner, works by running a Linux container in the background and enabling you to call into that container.
+
+Because of the foundational differences, compatibility with programs should be better in WSL2.
+
+[The way to enable them is the same between both versions](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+
+You'll simply want to run the following command in PowerShell as an administrator:
+
+```powershell
+Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
+```
+
+Then reboot your machine. After a reboot, you should be able to search for a Linux distro via the Microsoft Store. There are different options available to you, such as:
+
+- [Debian](https://www.microsoft.com/en-us/p/debian/9msvkqc78pk6)
+- [Kali Linux](https://www.microsoft.com/en-us/p/kali-linux/9pkr34tncv07)
+- [openSUSE](https://www.microsoft.com/en-us/p/opensuse-leap-15-1/9njfzk00fgkv)
+- [Ubuntu](https://www.microsoft.com/en-us/p/ubuntu/9nblggh4msv6)
+
+There are more distros than just those and more to come in the future.
+
+Once done, you simply run `bash` in a terminal Window or search for the distro name as an app. Either way should start-up the distro's instance of `bash` in the directory you were currently in. This is a full installation of Linux, meaning that you're able to access its package manager, run programs from it, and modify Windows files.
+
+There are even tweaks that're done with Windows to make it easier to use. If you run the `code` command to open a file, it will download a "VSCode remote server" to allow you to use your Windows install for both Linux and Windows commands. You can even [share SSH keys between Windows and WSL](https://devblogs.microsoft.com/commandline/sharing-ssh-keys-between-windows-and-wsl-2/)!
```

Comment:
```suggestion
There are even tweaks that are done with Windows to make it easier to use. If you run the `code` command to open a file, it will download a "VSCode remote server" to allow you to use your Windows install for both Linux and Windows commands. You can even [share SSH keys between Windows and WSL](https://devblogs.microsoft.com/commandline/sharing-ssh-keys-between-windows-and-wsl-2/)!
```